nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestElasticSearchQuery.java

```diff
@@ -229,4 +226,22 @@ public void testSecurity() throws Exception {
         }
     }
 
+    protected void createDocumentWithFiles() throws Exception {
+        startTransaction();
+        DocumentModel doc = session.createDocumentModel("/", "myFile", "File");
+        // create doc with a list of blob attachement textfile1.txt length=1,textfile2.txt length=2, ...
+        List<Map<String, Serializable>> blobs = new ArrayList<>();
+        for (int i = 1; i <= 5; i++) {
+            String name = "testfile" + i + ".txt";
+            Blob blob = Blobs.createBlob(new String(new char[i]).replace('\0', 'a'));
+            blob.setFilename(name);
+            blobs.add(Map.of("file", (Serializable) blob));
+        }
+        doc.setPropertyValue("files:files", (Serializable) blobs);
```

Comment:
You could use `var` keyword for blobs declaration and get rid of `files:files` cast.